python/activator/middleware_interface.py

```diff
@@ -1370,6 +1372,11 @@ def clean_local_repo(self, exposure_ids: set[int]) -> None:
                     self.butler.registry.removeCollection(member)
                 self.butler.removeRuns(calib_runs, unstore=True)
 
+            # TODO: workaround for DM-40193
```

Comment:
If this moved up to just after the `refresh()`, couldn't you put a `return` after it and skip all the other work, while still being nearly as compact code-wise?  But I guess it isn't really that much extra work, and in fact getting rid of other dataset types might speed up the `...` query.

Reply:
Nope, doesn't work -- I don't know if it's extra registry entries or something else, but the rearranged order leaves  about 190 MB in the pods instead of 2 MB. I'm worried that may grow in time.